ambry-utils/src/main/java/com.github.ambry.utils/Utils.java

```diff
@@ -682,4 +680,17 @@ public static ByteBuffer ensureCapacity(ByteBuffer existingBuffer, int newLength
     }
     return existingBuffer;
   }
+
+  /**
+   * Gets the root cause for {@code t}.
+   * @param t the {@link Throwable} whose root cause is required.
+   * @return the root cause for {@code t}.
+   */
+  public static Throwable getRootCause(Throwable t) {
```

Comment:
I see.  That makes sense.  I see now that some of the code passes in a Throwable, so it probably is a good idea to keep it as general as possible in case some other caller has a non-exception Throwable in the future.